contracts/QuestFactory.sol

```diff
@@ -77,9 +80,11 @@ contract QuestFactory is Initializable, OwnableUpgradeable, AccessControlUpgrade
         string memory contractType_,
         string memory questId_
     ) external onlyRole(CREATE_QUEST_ROLE) returns (address) {
-        if (quests[questId_].questAddress != address(0)) revert QuestIdUsed();
+        Quest storage currentQuest = quests[questId_];
+        if (currentQuest.questAddress != address(0)) revert QuestIdUsed();
+        ++currentQuest.numberMinted;
```

Comment:
This should not be added here! Causes `numberMinted` to start from 1 instead of 0